Add terms to glossary

slamoree-expertsupport opened the pull request:
## Hey, I just made a Pull Request!

Added terms to glossary

#### :heavy_check_mark: Checklist

<!--- Please include the following in your Pull Request when applicable: -->

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [X] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

Parsifal-M commented on docs/references/glossary.md:
```diff
@@ -139,11 +155,13 @@ A write stream of [documents](#documents).
-## JWT
+## JSON Web Token
 
-JSON Web Token.
+A popular, compact, and self-contained open standard for securely transmitting information between parties as a JSON object. It is commonly used to verify a user's identity [authentication](#authentication) and permissions [authorization](#authorization). Each JWT consists of a header, payload, and digital signature separated by dots. While often encrypted, JWTs are always signed for integrity. T ...
 
-A popular JSON based token format that is commonly encrypted and/or signed, see [the Wikipedia article](https://en.wikipedia.org/wiki/JSON_Web_Token) for more details.
+## JWT
+
+[JSON Web Token](#json-web-token).
```
I think we can merge these two perhaps? JSON Web Token (JWT) or AKA JWT? Something like this? 💭

Parsifal-M reviewed (commented):
Love it! Just a thought on the JWT part but overall I think its nice!

awanlin commented:
Hi @slamoree-expertsupport, if you can rebase that should get the build passing for you and we can try and merge this in 👍

slamoree-expertsupport commented:
Hmm. I had forgotten to sign the commit, so I did a pull to sync my local copy, rebased the branch, fixed a couple of sync errors, issued a signed commit, and pushed the branch. What did I miss?

Parsifal-M commented:
Am on mobile, but I think you just need to do a `git commit --amend --signoff` as it looks like your last commit is missing it 🤔

But I am no git surgeon 🤣

slamoree-expertsupport commented:
Thanks, Peter. That’s new to me. I’ll give it a try,Sent from my iPhoneOn Jun 1, 2025, at 3:46 AM, Peter Macdonald ***@***.***> wrote:﻿Parsifal-M left a comment (backstage/backstage#30037)
Am on mobile, but I think you just need to do a git commit --amend --signoff as it looks like your last commit is missing it 🤔
But I am no git surgeon 🤣

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

slamoree-expertsupport commented:
Rebased and pushed changes as instructed. Branch merged into my local master successfully.

Parsifal-M commented:
> Am on mobile, but I think you just need to do a `git commit --amend --signoff` as it looks like your last commit is missing it 

Hey @slamoree-expertsupport,

I think a commit is still missing a sign off for the DCO check, I think you might need to run the command above, more detailed information here that might be more useful --> https://github.com/backstage/backstage/pull/30037/checks?check_run_id=43629083012

Let us know if you'd like a hand!

slamoree-expertsupport commented:
Hi Peter,

Hmm, sorry for the bother. After receiving your earlier message, I tried
git commit —-amend —-signoff

Then, following the instructions in the referenced doc, I rebased the branch as follows:
git checkout add-terms-to-glossary
git rebase HEAD —-signoff
git push —-force-with-lease origin add-terms-to-glossary

It’s possible that I rebased to the wrong commit. I just rebased to what I think is the original signed commit, and then repeated the push.

If that doesn’t work, I may need help from someone how has more git skills than I have. (Thanks for the offer.)

Best,
Steve




> On Jun 10, 2025, at 3:21 AM, Peter Macdonald ***@***.***> wrote:
> 
> 
> Parsifal-M
>  left a comment 
> (backstage/backstage#30037)
>  <https://github.com/backstage/backstage/pull/30037#issuecomment-2958552173>
> Am on mobile, but I think you just need to do a git commit --amend --signoff as it looks like your last commit is missing it
> 
> Hey @slamoree-expertsupport <https://github.com/slamoree-expertsupport>,
> 
> I think a commit is still missing a sign off for the DCO check, I think you might need to run the command above, more detailed information here that might be more useful --> https://github.com/backstage/backstage/pull/30037/checks?check_run_id=43629083012 <https://github.com/backstage/backstage/pull/30037/checks?check_run_id=43629083012>
> Let us know if you'd like a hand!
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/backstage/backstage/pull/30037#issuecomment-2958552173>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/A2MTFBV2TN2LL7E6MJ5O36L3C2WTRAVCNFSM6AAAAAB5XKIIBWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDSNJYGU2TEMJXGM>.
> You are receiving this because you were mentioned.
>

benjdlambert commented:
@slamoree-expertsupport there's some custom instructions in the error check that might help you:


>To add your Signed-off-by line to every commit in this branch:
>Ensure you have a local copy of your branch by [checking out the pull request locally via command line](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/checking-out-pull-requests-locally).
>In your local branch, run: git rebase HEAD~21 --signoff
>Force push your changes to overwrite the branch: git push --force-with-lease origin add-terms-to-glossary

I think you might only be rebasing 1 commit which is why the `HEAD~21` in the above is important :)

slamoree-expertsupport commented:
Hi Ben,

You’re right—I didn’t realize that “HEAD~21” was special. I’ve followed your instructions more precisely. Let’s see if that fixes the problem.

Thanks for your advice, and for taking the time.

Best,
Steve

> On Jun 11, 2025, at 12:30 AM, Ben Lambert ***@***.***> wrote:
> 
> 
> benjdlambert
>  left a comment 
> (backstage/backstage#30037)
>  <https://github.com/backstage/backstage/pull/30037#issuecomment-2961550510>
> @slamoree-expertsupport <https://github.com/slamoree-expertsupport> there's some custom instructions in the error check that might help you:
> 
> To add your Signed-off-by line to every commit in this branch:
> Ensure you have a local copy of your branch by checking out the pull request locally via command line <https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/checking-out-pull-requests-locally>.
> In your local branch, run: git rebase HEAD~21 --signoff
> Force push your changes to overwrite the branch: git push --force-with-lease origin add-terms-to-glossary
> 
> I think you might only be rebasing 1 commit which is why the HEAD~21 in the above is important :)
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/backstage/backstage/pull/30037#issuecomment-2961550510>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/A2MTFBXY5YFBLVCTCME7N733C7LH7AVCNFSM6AAAAAB5XKIIBWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDSNRRGU2TANJRGA>.
> You are receiving this because you were mentioned.
>

slamoree-expertsupport commented:
As far as I know, this is ready for review. Please let me know if I'm mistaken.

slamoree-expertsupport commented:
Closed pull request. Superseded by https://github.com/backstage/backstage/pull/30346.
